From 9643a6ab2882f9b54458447562f933f51e68efcc Mon Sep 17 00:00:00 2001
From: Waldemar Rymarkiewicz <waldemarx.rymarkiewicz@intel.com>
Date: Mon, 23 Jul 2018 17:06:25 +0200
Subject: [PATCH] DRVLIB_SW-728: mips: Revert arch_irq_work_has_interrupt

Remove irq_work.h and let the system to use default
arch_irq_work_has_interrupt().

Irq_work for mips is not implemented, so we cannot return true here. Before
anyone else will enable this arch call there must be also arch_irq_work_raise()
implemented.

Signed-off-by: Waldemar Rymarkiewicz <waldemarx.rymarkiewicz@intel.com>
---
 arch/mips/Kbuild                 |  2 --
 arch/mips/include/asm/irq_work.h | 13 -------------
 2 files changed, 15 deletions(-)

diff --git a/arch/mips/Kbuild b/arch/mips/Kbuild
index f1993963a704..5c3f688a5232 100644
--- a/arch/mips/Kbuild
+++ b/arch/mips/Kbuild
@@ -3,8 +3,6 @@
 # CFLAGS_<file.o> := -Wno-error
 subdir-ccflags-y := -Werror
 
-generic-y += irq_work.h
-
 # platform specific definitions
 include arch/mips/Kbuild.platforms
 obj-y := $(platform-y)
diff --git a/arch/mips/include/asm/irq_work.h b/arch/mips/include/asm/irq_work.h
deleted file mode 100644
index 812f901c1d96..000000000000
--- a/arch/mips/include/asm/irq_work.h
+++ /dev/null
@@ -1,13 +0,0 @@
-#ifndef _ASM_IRQ_WORK_H
-#define _ASM_IRQ_WORK_H
-
-static inline bool arch_irq_work_has_interrupt(void)
-{
-#ifndef CONFIG_SMP
-	return false;
-#else
-	return true;
-#endif
-}
-
-#endif /* _ASM_IRQ_WORK_H */
